subsys/net/ip/net_if.c

```diff
@@ -2992,8 +2992,28 @@ int net_if_set_promisc(struct net_if *iface)
 
 void net_if_unset_promisc(struct net_if *iface)
 {
+	enum net_l2_flags l2_flags = 0;
+	int ret;
+
 	NET_ASSERT(iface);
 
+	if (net_if_l2(iface)->get_flags) {
+		l2_flags = net_if_l2(iface)->get_flags(iface);
+	}
+
+	if (!(l2_flags & NET_L2_PROMISC_MODE)) {
+		return;
+	}
+
+#if defined(CONFIG_NET_L2_ETHERNET)
```

Comment:
Could you try using IS_ENABLED instead?

Reply:
This is currently not possible as the ETHERNET symbol is only available if CONFIG_NET_L2_ETHERNET is set. But I can refactor the functions a bit more in order to avoid duplication. I will send a new version.